python/pyiceberg/utils/datetime.py

```diff
@@ -97,7 +96,7 @@ def datetime_to_millis(dt: datetime) -> int:
         delta = dt - EPOCH_TIMESTAMPTZ
     else:
         delta = dt - EPOCH_TIMESTAMP
-    return (delta.days * 86400 + delta.seconds) * 1_000 + delta.microseconds // 1_000
+    return int(delta.total_seconds() * 1_000)
```

Comment:
I'm not a fan of using floating point math in these places. Was there something incorrect about the previous implementation?

Reply:
I think the last one is easier to read. 
```suggestion
    return (delta.days * 86400 + delta.seconds) * 1_000 + delta.microseconds // 1_000
```
I can revert it, I don't want to make anyone paranoid